geniusyield-market-maker/src/GeniusYield/MarketMaker/Strategies.hs

```diff
@@ -199,213 +219,222 @@ fixedSpreadVsMarketPriceStrategy
+
+      getOrdersLockedValue ∷ MMTokenPair → MMToken → [(MMTokenPair, PartialOrderInfo)] → Natural
+      getOrdersLockedValue mmtp st orders =
+        let relevantOfferedAc = mmtAc st
+            relevantOrders = filter (\(oStp, oPoi) → oStp == mmtp && relevantOfferedAc == poiOfferedAsset oPoi) orders
+         in sum $ map (poiOfferedAmount . snd) relevantOrders
+
+      logInfo, logDebug, logWarn ∷ GYProviders → String → IO ()
+      logInfo providers = gyLogInfo providers logNS
+      logDebug providers = gyLogDebug providers logNS
+      logWarn providers = gyLogWarning providers logNS
+
+      logPlaceAction ∷ PlaceOrderAction → String
+      logPlaceAction PlaceOrderAction {..} =
+        let price = (fromRational (rationalToGHC poaPrice) ∷ Double)
+            adjustedPrice = 1 / price
+         in unwords
+              [ "Selling",
+                show poaOfferedAmount,
+                prettyAc poaOfferedAsset,
+                "for",
+                show price,
+                prettyAc poaAskedAsset,
+                "each",
+                "(inverted price:",
+                show adjustedPrice,
+                ")"
+              ]
+      logMarketInfo ∷ Price → String
+      logMarketInfo price =
+        unwords
+          [ "Price for:",
+            prettyAc $ mmtAc mmToken,
+            "is",
+            show (fromRational (getPrice price) ∷ Double),
+            "lovelaces"
+          ]
+
+      prettyAc ∷ GYAssetClass → String
+      prettyAc GYLovelace     = "lovelaces"
+      prettyAc (GYToken _ tn) = "indivisible of " ++ show tn
+
+ordersToBeRemoved ∷ Price → Spread → [(MMTokenPair, PartialOrderInfo)] → [(MMTokenPair, PartialOrderInfo)]
 ordersToBeRemoved price cancelLimitSpread = filter (orderIsToBeRemoved price cancelLimitSpread)
 
-orderIsToBeRemoved :: Price -> Spread -> (MMTokenPair, PartialOrderInfo) -> Bool
+orderIsToBeRemoved ∷ Price → Spread → (MMTokenPair, PartialOrderInfo) → Bool
 orderIsToBeRemoved mPrice _cancelLimitSpread@Spread {..} (mmtp, poi) =
   let marketPrice = getPrice mPrice
       oap = toOAPair mmtp
    in case mkOrderInfo oap poi of
-        SomeOrderInfo OrderInfo {orderType = SBuyOrder, price} -> getPrice price < marketPrice - (buySideSpread * marketPrice)
-        SomeOrderInfo OrderInfo {orderType = SSellOrder, price} -> getPrice price > marketPrice + (sellSideSpread * marketPrice)
+        SomeOrderInfo OrderInfo {orderType = SBuyOrder, price} → getPrice price < marketPrice - (buySideSpread * marketPrice)
+        SomeOrderInfo OrderInfo {orderType = SSellOrder, price} → getPrice price > marketPrice + (sellSideSpread * marketPrice)
+
+flip4 ∷ (a → b → c → d → r) → (b → c → d → a → r)
+flip4 f = \b c d a → f a b c d
+
+logPricesProviderFail ∷ GYProviders → PriceIndicator → IO ()
+logPricesProviderFail providers pI = case pI of
+  PriceSourceFail es pps _ → do
+    gyLogWarning providers logNS $ "Some prices provider(s) failed: " ++ (show pps)
+    mapM_ (gyLogInfo providers logNS) es
```

Comment:
Since we are in `gyLogWarning`, and it seems like a continuation of same message, let's also have it under warning severity.

Reply:
Done.